### PR TITLE
Implement #1492: basic Vector<Float> support with Number arrays

### LINKED ISSUE #1492: Data API JsonCodec: Vector, add support for `vector<float>`
As distinct from set, list, need to add support for "vector" type: but only with `float` elements, that is, `vector<float>`.

Initial implementation to take in "simple" or "plain" JSON number array; same as Collections. Will file separate issue for Blob/Base64-encoded variant.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds support for Vector type (reading, writing) to/from JSON Array representation

**Which issue(s) this PR fixes**:
Fixes #1492

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistryTest.java [hunk 1]
```diff
@@ -639,28 +679,13 @@ public void unknownColumnException() {
...
-              assertThat(e.targetCQLType).isEqualTo(typeToTest);
-              assertThat(e.value).isEqualTo(valueToTest);
-
-              assertThat(e.getMessage())
-                  .contains(typeToTest.toString())
-                  .contains(valueToTest.getClass().getName())
-                  .contains(valueToTest.toString())
-                  .contains("Root cause: " + rootCause);
-            });
+    assertToCQLFail(
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistryTest.java:682]: This (and a few following changes) is unrelated refactoring to reduce verbosity of unit tests here -- no functional changes.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java [hunk 2]
```diff
@@ -116,6 +117,17 @@ public <JavaT, CqlT> JSONCodec<JavaT, CqlT> codecToCQL(
               CollectionCodecs.buildToCQLSetCodec(valueCodecCandidates, st.getElementType());
         }
         // fall through
+      } else if (columnType instanceof VectorType vt) {
```
Comment (Yuqi-Du) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java:120]: https://cassandra.apache.org/doc/latest/cassandra/developing/cql/types.html#collections
One thing is interesting, c* does not call vector as a collection column dataType.
But vector is not a scalar.
So it is kind of stuck in the middle.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            (objectMapper, cqlType, value) -> toJsonNode(objectMapper, (CqlVector<Number>) value));
+  }
+
+  static CqlVector<Float> toCQLFloatVector(VectorType vectorType, Collection<?> listValue)
+      throws ToCQLCodecException {
+    Collection<JsonLiteral<?>> vectorIn = (Collection<JsonLiteral<?>>) listValue;
+    final int expLen = vectorType.getDimensions();
+    if (expLen != vectorIn.size()) {
+      throw new ToCQLCodecException(
+          vectorIn,
```
Comment (Yuqi-Du) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java:52]: Just wild thinking, does this mean, the error msg has a long user-input vector in the response?
Do we want to just let user know there is a dimension mismatch.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    }
+    List<Float> floats = new ArrayList<>(expLen);
+    for (JsonLiteral<?> literalElement : vectorIn) {
+      Object element = literalElement.value();
+      if (element instanceof Number num) {
+        floats.add(num.floatValue());
+        continue;
+      }
+      throw new ToCQLCodecException(
+          vectorIn,
```
Comment (Yuqi-Du) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java:64]: Same here? (ignore this)

Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java:64]: Oh, this is inside of the for loop. 

Nit, Not sure if '#x' is clear to indicate this is referring position.

**Review by Yuqi-Du — APPROVED** [review]
Approved, some comments.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java (hunk 2 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java:120]: Yes, need to keep that in mind wrt terminology. LMK if I missed some reference, calling vectors collections?

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java (hunk 3 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java:52]: I think this needs to be handled generically, to limit output length for all kinds of input values: I think `ToCQLCodecException` expects input value to be provided.
I'll see if `ToCQLCodecException` limits length yet.

Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java:52]: Ok: value was NOT yet truncated -- added code to truncate to 1000 chars if longer.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java (hunk 4 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java:64]: Ok I'll try to improve message a bit.

**Review by vkarpov15 — COMMENTED** [review]
LGTM :+1:
